### DISCUSSION TITLE
Fixed Toolbar, Sticky Toolbar

### DISCUSSION THREAD

**Discussion by saurav2799** [codex-team/editor.js#1943] [Ideas]
1. Describe a problem.
We are looking to have option of fixed toolbar. This is particularly useful for less internet savvy users, so they can identify the toolbar and format options. so far our team was unable to find this option in the docs / demos (env: reactjs setup)

bonus: if toolbar can stick at top of the content box/div, so users can scroll comfortably on long documents.

2. Describe the solution you'd like. Mockups are welcome.
for example, we have in this submission: 
![image](https://user-images.githubusercontent.com/52714019/132300466-3527c735-6232-4189-aeff-5a19d880032f.png)

3. Are there any alternatives?
for other alternatives, we might have to consider alternate editor - we have looked at tiny, draftjs and now evaluating slatejs.

**Comment by hor4z** [comment]
I am also really interested in this feature, any news?, is it possible to achieve this currently?

**Comment by WowItsDoge** [comment]
This would be a realy nice feature.

**Comment by hor4z** [comment]
any new about that?

**Comment by templeman15** [comment]
We are very interested in this, about the only reason we are undecided on using editor js as we need the top toolbar!

**Reply by Mowmowj** [reply]
Is this toolbar easy to write by ourselves? I wanna write a plugin to achieve it, Is there anybody interested in it?

**Reply by templeman15** [reply]
We weren't able to figure it out without a hacky solution. We ended up switching to [tiptap](https://tiptap.dev/).

**Reply by sgebr01** [reply]
Tiptap seems even better, thank you for letting us know.

**Comment by Mowmowj** [comment]
Is this toolbar easy to write by ourselves? 
I wanna write a plugin to achieve it. 
Is there anybody interested in it?

**Reply by ieliwb** [reply]
Hello, is it completed?

**Comment by intaek-h** [comment]
waiting for an update..

**Comment by aske-cph** [comment]
Is this project abandoned? I'm also switching to TipTap unfortunately. I find the lack of toolbar pretty weird since it's such a basic a feature, also seems like many people are asking this question here and on Stack.

**Comment by sbesic-starke** [comment]
Unbelieveable. Such a basic functionality. Switching to TipTap.

**Comment by gauravsaluja2006** [comment]
Would love to be able to have a fixed top toolbar.

**Comment by miketalley** [comment]
Bumping this as I would love to use EditorJS for our project, but it will be hard to convince the team to use it with no ability to have a fixed toolbar displayed.

**Comment by ieliwb** [comment]
any new about that?

**Comment by jiangyuhang0506** [comment]
Will such a top fixed toolbar plugin be provided？
I am looking forward to this feature very much, and it will be great！！

**Comment by tiagozip** [comment]
Bump. If this was optional and customizable it would be pretty nice. But at least for me this isn't a priority.

Also this was made in 2021. It's now 2024.

**Comment by jakeandreoli** [comment]
+1 would be absolutely wonderful addition

**Comment by tasouza** [comment]
Is this not possible to create using the API?

Is the editorjs API complete and simple to the point that I can detect selected parts and apply formatting only to that part? because if it is possible, I believe that creating a fixed toolbar would be easy.

**Comment by tiagozip** [comment]
I don't really think this is being maintained enough for this to actually get into Editor.js

**Comment by rastermechanism** [comment]
Hi, I have been working with with many editors, I would advoate opensource always.  My requirement was more skewed towards manipulation of different typset. Initially I was using tiny but they turnout to be looters, they asked my to link the card and day I linked my  they deducted $2500, before everything was under free 1K hit. Seems those company are making fortune. 

![Screenshot 2024-07-17 154231](https://github.com/user-attachments/assets/6ad572dd-cb0f-43d1-b420-d4b168324d9d)
Anyway regarding fixed toolbar. I have simulated some of it. For now just have created custom tool bar with plain text. later will move to svg icon and have added few. Here is one example with attachment. Once I have everything working will update you. However just for proof of concept here is what I have tried.

**Comment by dons20** [comment]
Has anyone from the editor.js team actually responded to this request? It seems like there's no interest in bringing this to reality which is a shame if true.

**Comment by RealMrHex** [comment]
sadlyyyy, switched to tiptap :(

**Reply by cristianmatos** [reply]
Yes, I'm switching too.
